erigon-lib/sysutils/sysutils.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2024 The Erigon Authors
+// This file is part of Erigon.
+//
+// Erigon is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Erigon is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Erigon. If not, see <http://www.gnu.org/licenses/>.
+
+package sysutils
+
+import (
+	"time"
+
+	"github.com/erigontech/erigon-lib/log/v3"
+	"github.com/shirou/gopsutil/v4/process"
+)
+
+type ProcessInfo struct {
+	Pid      int32
+	Name     string
+	CPUUsage float64
+	Memory   float32
+}
+
+type ProcessMerge struct {
+	CPUUsage float64
+	Memory   float32
+	Times    int
+	Name     string
+}
+
+const (
+	iterations     = 5
+	sleepSeconds   = 2
+	usageThreshold = 0.05
+)
+
+func GetProcessesInfo() []*ProcessInfo {
+	procs, err := process.Processes()
+	if err != nil {
+		log.Debug("[Sysutil] Error retrieving processes: %v", err)
+	}
+
+	return averageProceses(procs)
+}
+
+func AverageProceses(procs []*process.Process) []*ProcessInfo {
+	return averageProceses(procs)
+}
+
+func averageProceses(procs []*process.Process) []*ProcessInfo {
+	// Collect processes and calculate average stats.
+	allProcsRepeats := make([][]*ProcessInfo, 0, iterations)
+
+	// Collect all processes N times with a delay of N seconds to calculate average stats.
+	for i := 0; i < iterations; i++ {
+		processes := allProcesses(procs)
+		allProcsRepeats = append(allProcsRepeats, processes)
+		time.Sleep(sleepSeconds * time.Second)
+	}
+
+	// Calculate average stats.
+	averageProcs := mergeProcesses(allProcsRepeats)
+	averageProcs = removeProcessesAboveThreshold(averageProcs, usageThreshold)
+
+	return averageProcs
+}
+
+func RemoveProcessesAboveThreshold(processes []*ProcessInfo, treshold float64) []*ProcessInfo {
```

Comment:
should these 2 funcs be named `RemoveProcessesBelowThreshold` instead of `Above`?